test/mpi/test_mpi.cc

```diff
@@ -362,6 +362,15 @@ bool ucc_coll_has_datatype(ucc_coll_type_t c)
     case UCC_COLL_TYPE_REDUCE:
     case UCC_COLL_TYPE_REDUCE_SCATTER:
```

Comment:
so, basically now ALL colls except for Barrier,Fanin,Fanout have dt, right? maybe re-write the check with smaller number of "cases"?

Reply:
done